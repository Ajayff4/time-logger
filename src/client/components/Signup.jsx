import React, { Component } from 'react';
import axios from 'axios';
import sha256 from 'sha256';
import { signup } from '../actions';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types'

class Signup extends Component {
    state = {
        username: "",
        fullname: "",
        email: "",
        password: "",
        repassword: ""
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit = (e) => {
        e.preventDefault();
        const { username, fullname, email, password, repassword } = this.state;
        if (password === repassword) {
            axios.post('/signup', {
                fullname: fullname,
                username: username,
                email: email,
                password: sha256(password)
            }).then(res => {
                console.log(res.status);
                alert("ok");
                this.props.signup();
                this.props.history.push('/login');
            }).catch(err => {
                console.error("post: ", err);
            });
        }
    }

    render() {
        return (
            <center>
                <div>
                    <fieldset>
                        <legend>Signup</legend>
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="username"
                                id="username"
                                placeholder="username"
                                value={this.state.username}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="text"
                                name="fullname"
                                id="fullname"
                                placeholder="fullname"
                                value={this.state.fullname}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="email"
                                name="email"
                                id="email"
                                placeholder="email"
                                value={this.state.email}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="password"
                                name="password"
                                id="password"
                                placeholder="password"
                                value={this.state.password}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="password"
                                name="repassword"
                                id="repassword"
                                placeholder="repeat password"
                                value={this.state.repassword}
                                onChange={this.onChange}
                                required
                            /><br />
                            <button type="submit">Sign Up</button>
                        </form>
                    </fieldset>
                </div>
            </center>
        )
    }
}

Signup.propTypes = {
    signup: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => {
    return {
        signup: () => dispatch(signup())
    }
}

export default connect(null, mapDispatchToProps)(withRouter(Signup));