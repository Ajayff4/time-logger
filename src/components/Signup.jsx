import React, { Component } from 'react'
import axios from 'axios'
import * as actions from '../actions'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import Toast from './Toast'

class Signup extends Component {
    static propTypes = {
        signup: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }

    state = {
        username: "",
        fullname: "",
        email: "",
        password: "",
        repassword: "",
        toast: {
            status: false,
            title: "",
            type: "",
            message: ""
        }
    }

    renderToast = () => {
        setTimeout(() => this.setState({toast: false}),3000)
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault()
        const { username, fullname, email, password, repassword } = this.state
       
        if (password === repassword) {
            axios.post('http://localhost:5000/rest/users/signup',{
                username: username,
                fullname: fullname,
                email: email,
                password: password
            },{
                headers: {
                    "Access-Control-Allow-Origin": "*"
                }
            })
            .then(res => {
                this.props.signup()
                this.setState({
                    ...this.state,
                    toast: {
                        ...this.state.toast,
                        status: true,
                        title: "Signup",
                        type: "toastSuccess",
                        message: "Signup Successful."
                    }
                })
                this.renderToast()
                this.props.history.push('/login')
            }).catch(err => {
                this.setState({
                    ...this.state.toast,
                    status:true,
                    title: "Signup",
                    type: "toastSuccess",
                    message: "Signup Successful."
                })
                this.renderToast()
                this.props.failedToFetchData(`${err}`);
            })
        }
    }

    render() {
        return (
            <center>
                <div>
                    {this.state.toast.status? 
                        <Toast
                            title={this.state.toast.title}
                            type={this.state.toast.type}
                            message={this.state.toast.message} 
                        />
                    :null}
                    <fieldset>
                        <legend>Signup</legend>
                        <form 
                            method="POST" 
                            onSubmit={this.onSubmit} 
                            autoComplete="off"
                        >
                            <input
                                type="text"
                                name="username"
                                id="username"
                                placeholder="username"
                                value={this.state.username}
                                onChange={this.onChange}
                                required
                            /><br />
                            
                            <input
                                type="text"
                                name="fullname"
                                id="fullname"
                                placeholder="fullname"
                                value={this.state.fullname}
                                onChange={this.onChange}
                                required
                                autoComplete="off"
                            /><br />
            
                            <input
                                type="email"
                                name="email"
                                id="email"
                                placeholder="email"
                                value={this.state.email}
                                onChange={this.onChange}
                                required
                                autoComplete="off"
                            /><br />

                            <input
                                type="password"
                                name="password"
                                id="password"
                                placeholder="password"
                                value={this.state.password}
                                onChange={this.onChange}
                                required
                                style={{borderBlockColor: 'lime'}}
                            /><br />

                            <input
                                type="password"
                                name="repassword"
                                id="repassword"
                                placeholder="repeat password"
                                value={this.state.repassword}
                                onChange={this.onChange}
                                required
                            /><br />

                            <button type="submit">Sign Up</button>
                        </form>
                    </fieldset>
                </div>
            </center>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        failedToFetchData: (error) => dispatch(actions.failedToFetchData(error)),
        signup: () => dispatch(actions.signup())
    }
}

export default connect(null, mapDispatchToProps)(withRouter(Signup))