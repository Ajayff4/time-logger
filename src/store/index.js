import { createStore, compose, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    compose(
        applyMiddleware(sagaMiddleware),
        window.__REDUX_DEVTOOLS_EXTENSION__ &&
        window.__REDUX_DEVTOOLS_EXTENSION__()
    ),
)

sagaMiddleware.run(rootSaga)

export default store