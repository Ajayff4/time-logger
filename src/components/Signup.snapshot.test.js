import React from 'react'
import renderer from 'react-test-renderer'
import Signup from './Signup'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Signup Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Signup /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});