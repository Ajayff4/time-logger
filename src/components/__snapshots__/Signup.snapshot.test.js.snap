// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Snapshot Testing... renders component correctly 1`] = `
<center>
  <div>
    <fieldset>
      <legend>
        Signup
      </legend>
      <form
        autoComplete="off"
        method="POST"
        onSubmit={[Function]}
      >
        <input
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="username"
          required={true}
          type="text"
          value=""
        />
        <br />
        <input
          autoComplete="off"
          id="fullname"
          name="fullname"
          onChange={[Function]}
          placeholder="fullname"
          required={true}
          type="text"
          value=""
        />
        <br />
        <input
          autoComplete="off"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
        <br />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="password"
          required={true}
          style={
            Object {
              "borderBlockColor": "lime",
            }
          }
          type="password"
          value=""
        />
        <br />
        <input
          id="repassword"
          name="repassword"
          onChange={[Function]}
          placeholder="repeat password"
          required={true}
          type="password"
          value=""
        />
        <br />
        <button
          type="submit"
        >
          Sign Up
        </button>
      </form>
    </fieldset>
  </div>
</center>
`;
